### PR TITLE
FW position control: setpoint vx,vy,z

### PR DESCRIPTION
**Pull request by korotkoves**
Please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io/) to align on pull requests if necessary. You can then open draft pull requests to get early feedback.

**Describe problem solved by this pull request**
In offboard mode fixedwings/vtols can be controlled only by waypoints.

**Describe your solution**
Add possibility to control with horizontal velocity vector (vx,vy) and height (z).

**Describe possible alternatives**
Unknown

**Test data / coverage**
Gazebo simulation.

### CODE REVIEW

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 1]
```diff
@@ -904,7 +891,24 @@ FixedwingPositionControl::control_position(const hrt_abstime &now, const Vector2
 				_tecs.set_height_error_time_constant(_param_fw_thrtc_sc.get() * _param_fw_t_h_error_tc.get());
 			}
 
-			tecs_update_pitch_throttle(now, alt_sp,
+		} else if (pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_VELOCITY) {
```
Comment (dagar) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:894]: The position setpoint SETPOINT_TYPE_VELOCITY will be gone soon, can you entirely use `trajectory_setpoint`?

https://github.com/PX4/PX4-Autopilot/pull/17029

**Comment by dagar** [comment]
FYI @Jaeyoung-Lim

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 1 above)
Comment (korotkoves) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:894]: If we remove SETPOINT_TYPE_VELOCITY then all logic in  control_position() depends on waypoints.
We can use yaw and yaw_valid in position_setpoint.msg to differentiate velocity control mode but must do many checks across the code in control_position() and logic will be not simple.

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -1014,14 +1014,13 @@ MavlinkReceiver::handle_message_set_position_target_global_int(mavlink_message_t
 
 		const uint16_t type_mask = target_global_int.type_mask;
 
-		// position
-		if (!(type_mask & (POSITION_TARGET_TYPEMASK_X_IGNORE | POSITION_TARGET_TYPEMASK_Y_IGNORE |
-				   POSITION_TARGET_TYPEMASK_Z_IGNORE))) {
+		// position x,y
+		if (!(type_mask & (POSITION_TARGET_TYPEMASK_X_IGNORE | POSITION_TARGET_TYPEMASK_Y_IGNORE))) {
```
Comment (Jaeyoung-Lim) [inline src/modules/mavlink/mavlink_receiver.cpp:1018]: Has the case where position setpoints are sent, but either xy or z is ignored without velocity setpoints?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 3]
```diff
@@ -1033,6 +1032,21 @@ MavlinkReceiver::handle_message_set_position_target_global_int(mavlink_message_t
 			const double lon = target_global_int.lon_int / 1e7;
 			map_projection_project(&global_local_proj_ref, lat, lon, &setpoint.x, &setpoint.y);
 
+		} else {
+			setpoint.x = NAN;
+			setpoint.y = NAN;
+		}
+
+		// position z
+		if (!(type_mask & (POSITION_TARGET_TYPEMASK_Z_IGNORE))) {
```
Comment (Jaeyoung-Lim) [inline src/modules/mavlink/mavlink_receiver.cpp:1041]: Currently this is only done for `SET_TARGET_POSITION_GLOBAL_INT`. we need to do the same for `SET_TARGET_POSITION_LOCAL_NED`

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 4]
```diff
@@ -913,12 +917,6 @@ FixedwingPositionControl::control_position(const hrt_abstime &now, const Vector2
 						   tecs_fw_mission_throttle,
 						   false,
 						   radians(_param_fw_p_lim_min.get()));
-
-		} else if (position_sp_type == position_setpoint_s::SETPOINT_TYPE_LAND) {
-			control_landing(now, curr_pos, ground_speed, pos_sp_prev, pos_sp_curr);
-
-		} else if (position_sp_type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF) {
-			control_takeoff(now, curr_pos, ground_speed, pos_sp_prev, pos_sp_curr);
```
Comment (Jaeyoung-Lim) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:921]: Could you explain why this was removed?

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 1 above)
Comment (Jaeyoung-Lim) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:894]: @korotkoves The plan is to make everything depend on `trajectory_setpoint` and not waypoints

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
> Test data / coverage
> Gazebo simulation.

Could you be more specific on what was tested, how(mavsdk / ROS etc) and what the results were? (logs .etc)

The type mask changes introduce the following cases that needs to be tested. 
- position_xy
- position_xy + position_z
- velocity_xy
- velocity_xy + position_z
- position_z

This is also enabling `vel_xy + pos_z` case for multirotors. We need to test this to make sure we aren't introducing any unexpected behaviors

**Comment by Jaeyoung-Lim** [comment]
Since this PR has become stale, and in favor of https://github.com/PX4/PX4-Autopilot/pull/18495 closing this PR

**Pull request closed without merging**